### PR TITLE
fix(agw): Modified code in sgw_s8 task to port ue context hash table to protobuf map

### PR DESCRIPTION
**Pull request by rsarwad**
## Summary
fix(agw): Modified code in sgw_s8 task to port ue context hash table to protobuf map
Removed dependency of spgw task and sgw_s8 task on hash table
## Test Plan
Executed unit test cases and built with "make build_oai" and executed s1ap sanity test suite
We will not be able to verify inbound roaming feature in local setup.
## Additional Information
Needs to be tested federated integ_tests once PCRF is integrated

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/sgw/spgw_state.cpp [hunk 1]
```diff
@@ -130,16 +130,21 @@ void sgw_free_ue_context(void** ptr) {
...
+  }
+  sgw_s11_teid_t* p1 = LIST_FIRST(&(ue_context_p)->sgw_s11_teid_list);
+  sgw_s11_teid_t* p2 = nullptr;
+  while (p1) {
+    p2 = LIST_NEXT(p1, entries);
+    LIST_REMOVE(p1, entries);
+    free_cpp_wrapper(reinterpret_cast<void**>(&p1));
+    p1 = p2;
   }
+  free_cpp_wrapper((void**)ptr);
```
[the remark this answers is not in the document]
Comment (pruthvihebbani) [reply lte/gateway/c/core/oai/tasks/sgw/spgw_state.cpp]: Please finx cpplint errors

File: lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_state_manager.cpp [hunk 2]
```diff
@@ -77,14 +74,10 @@ void SgwStateManager::create_state() {
...
-  state_cache_p->imsi_ue_context_htbl =
-      hashtable_ts_create(SGW_STATE_CONTEXT_HT_MAX_SIZE, nullptr,
-                          (void (*)(void**))sgw_free_ue_context, nullptr);
-  if (!(state_cache_p->imsi_ue_context_htbl)) {
-    OAILOG_CRITICAL(LOG_SGW_S8,
-                    "Failed to create imsi_ue_context_htbl for SGW_S8 task \n");
-    return;
-  }
+  state_cache_p->imsi_ue_context_map.map =
+      new google::protobuf::Map<unsigned long int, struct spgw_ue_context_s*>();
```
[the remark this answers is not in the document]
Comment (pruthvihebbani) [reply lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_state_manager.cpp]: Fix this error

**Review by pruthvihebbani — CHANGES_REQUESTED** [review]

**Review by pruthvihebbani — APPROVED** [review]
LGTM

File: lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp [hunk 3]
```diff
@@ -41,7 +41,6 @@ extern "C" {
 #include "lte/gateway/c/core/oai/lib/itti/intertask_interface_types.h"
 #include "lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u.h"
 #include "lte/gateway/c/core/oai/common/conversions.h"
-#include "lte/gateway/c/core/oai/lib/hashtable/hashtable.h"
 #include "lte/gateway/c/core/oai/lib/bstr/bstrlib.h"
 #include "lte/gateway/c/core/oai/lib/itti/itti_types.h"
 #ifdef __cplusplus
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp:48]: Please move this header https://github.com/magma/magma/blob/a49ee7522c670afb058d56023000e1c2c530435e/lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp#L71 to the top of the include list

File: lte/gateway/c/core/oai/test/sgw_s8_task/sgw_s8_utility.cpp [hunk 4]
```diff
@@ -218,8 +218,7 @@ bool is_num_s1_bearers_valid(sgw_state_t* sgw_state, imsi64_t imsi64,
                              int expected_num_active_bearers) {
   spgw_ue_context_t* ue_context_p = NULL;
```
Comment (ajahl) [inline lte/gateway/c/core/oai/test/sgw_s8_task/sgw_s8_utility.cpp:222]: can you move https://github.com/magma/magma/blob/a49ee7522c670afb058d56023000e1c2c530435e/lte/gateway/c/core/oai/test/sgw_s8_task/sgw_s8_utility.cpp#L15 to the top, please

File: lte/gateway/c/core/oai/test/sgw_s8_task/sgw_s8_utility.cpp [hunk 5]
```diff
@@ -218,8 +218,7 @@ bool is_num_s1_bearers_valid(sgw_state_t* sgw_state, imsi64_t imsi64,
...
```
Comment (ajahl) [inline lte/gateway/c/core/oai/test/sgw_s8_task/sgw_s8_utility.cpp]:
```cpp
spgw_ue_context_t* ue_context_p = nullptr;
```

File: lte/gateway/c/core/oai/test/sgw_s8_task/sgw_s8_utility.cpp [hunk 6]
```diff
@@ -218,8 +218,7 @@ bool is_num_s1_bearers_valid(sgw_state_t* sgw_state, imsi64_t imsi64,
...
 
-  hashtable_ts_get(sgw_state->imsi_ue_context_htbl, (const hash_key_t)imsi64,
-                   (void**)&ue_context_p);
+  sgw_state->imsi_ue_context_map.get(imsi64, &ue_context_p);
```
Comment (ajahl) [inline lte/gateway/c/core/oai/test/sgw_s8_task/sgw_s8_utility.cpp:224]: please change
https://github.com/magma/magma/blob/a49ee7522c670afb058d56023000e1c2c530435e/lte/gateway/c/core/oai/test/sgw_s8_task/sgw_s8_utility.cpp#L226
to
```cpp
sgw_s11_teid_t* s11_teid_p = nullptr;
```

**Review by ajahl — APPROVED** [review]

**Pull request merged**